salt/modules/tls.py

```diff
@@ -846,13 +847,17 @@ def create_ca_signed_cert(ca_name, CN, days=365, cacert_path=None, digest='sha25
     cert.set_issuer(ca_cert.get_subject())
     cert.set_pubkey(req.get_pubkey())
     extensions_list = []
-    for name in extensions:
-        log.debug("name: {0}, critical: {1}, options: {2}".format(
-            name, extensions[name]['critical'], extensions[name]['options']))
-        extensions_list.append(OpenSSL.crypto.X509Extension(
-            name,
-            extensions[name]['critical'],
-            extensions[name]['options']))
+    for name, edata in six.iteritems(extensions):
+        if not isinstance(edata, dict):
+            continue
+        for opt in ['critical', 'options']:
+            if opt not in edata:
+                break
+        else:
```

Comment:
No, the else is for the for loop. If the break is not hit during the for loop, that's when the extensions_list.append should be executed.